test/entity/multipleFieldsEntity.js

```diff
@@ -1,135 +1,163 @@
 const { entity } = require('../../src/entity')
 const { field } = require('../../src/field')
 const assert = require('assert')
+const { id } = require('../../src/id')
 
 describe('An entity', () => {
+  describe('with multiple fields', () => {
+    const NewEntity = entity('New Entity', {
+      f1: field(String),
+    })
+
+    const givenAnEntityWithMultipleFields = () => {
+      const AnEntity = entity('An entity', {
+        field1: field(Number),
+        field2: field(String),
+        field3: field(Date),
+        field4: field(Boolean),
+        field5: field(NewEntity),
+        field6: field([NewEntity]),
+        field7: id(Number),
+        field8: id(String),
+        field9: field(Number, { isId: true }),
+        field10: id(Date),
+        field11: field(Boolean, { isId: true }),
+      })
+      return new AnEntity()
+    }
+
+    it('should initiate', () => {
+      //given
+      const instance = givenAnEntityWithMultipleFields()
+      //then
+      assert.equal(instance.meta.name, 'An entity')
+    })
+
+    it('should set a value to multiple fields', () => {
+      //given
+      const instance = givenAnEntityWithMultipleFields()
+      //when
+      instance.field1 = 1
+      instance.field2 = '1'
+      instance.field3 = new Date('2019-09-30T23:45:34.324Z')
+      instance.field4 = true
+      const newEntity = new NewEntity()
+      newEntity.f1 = 'abc'
+      instance.field5 = newEntity
+      instance.field6 = [newEntity]
+      //then
+      assert.strictEqual(instance['field1'], 1)
+      assert.strictEqual(instance['field2'], '1')
+      assert.deepStrictEqual(
+        instance['field3'],
+        new Date('2019-09-30T23:45:34.324Z')
+      )
+      assert.strictEqual(instance['field4'], true)
+      assert.deepStrictEqual(instance['field5'], newEntity)
+      assert.deepStrictEqual(instance['field5'].f1, 'abc')
+      assert.deepStrictEqual(instance['field6'][0], newEntity)
+      assert.deepStrictEqual(instance['field6'][0].f1, 'abc')
+    })
+
+    it('should have multiple instances with isolated valued from each other', () => {
+      //given
+      const instance1 = givenAnEntityWithMultipleFields()
```

Comment:
why not use an array?
``` js
const instances = Promise.all(Array(3).map(x => givenAnEntityWithMultipleFields()))
instances[0]
instances[1]
instances[2]

Reply:
Sorry, my linter changed the hole file.. This test was not implemented by this pr, but hey, I can fix it on another pr, no problem